app/scripts/controllers/metametrics.js

```diff
@@ -283,15 +283,30 @@ export default class MetaMetricsController {
   }
 
   /**
-   * track a metametrics event, performing necessary payload manipulation and
+   * submits a metametrics event, not waiting for it to complete or allowing its error to bubble up
```

Comment:
Why is it desirable to not wait for it to complete? We can catch errors and prevent them from bubbling up without making that change.

Reply:
i think that metric event delivery should have no impact on the submitter, it should simply be something queued for delivery

Reply:
That probably makes sense in most cases, but the ability to wait until a metric event has been sent before continuing can be useful for events that are a part of a funnel. That's the only way to guarantee they're sent in the correct order. 

The Mixpanel funnels rely upon set event orderings to detect which events correspond to which funnel. It allows funnels to have events that are recieved in different orders in some cases, but the first and last event in the defined funnel would have to come first and last respectively, to ensure that those funnel events don't get mixed up with another funnel instance (e.g. in the case where the user goes through another identical flow, which isn't uncommon for sending, swaps, confirmations, etc.).

We do already call `trackEvent` without `await`-ing it in a few places where we know there's no reason to wait, and we can probably start doing that more often. But to make this function non-blocking would prevent us from waiting in the cases where it matters.

Reply:
i dont think you're ever supposed to have unhandled promises, so "simply not awaiting it" seems like a bad pattern

Reply:
added validation to `trackEvent`